src/jmap-email/jmap_email/composer.py

```
@@ -231,46 +344,153 @@ def _normalize_date(date) -> datetime.datetime:
+    """Join a JMAP ``MessageIds`` ``String[]`` into a single
+    space-separated chain (angle-bracket form) for the wire.
+
+    Strict-typed: see :func:`_first_msgid` — only accepts a list.
+    """
+    if not isinstance(value, list) or not value:
+        return ""
+    chain: list[str] = []
+    for v in value:
+        if not isinstance(v, str) or not v:
+            continue
+        v = v.strip()
```

Ruff (0.15.15)

[warning] `for` loop variable `v` overwritten by assignment target

(PLW2901)